libs/frontend/feature-site-translations/src/lib/use-site-translations.ts

```diff
@@ -51,8 +52,17 @@ export const useSiteTranslations = (): ISiteTranslationsFeature => {
   watch(
     () => editor.value?.translations,
     (show) => {
-      if (show) {
-        lang.value = getActiveI18n()
+      const { activeI18n, i18n } = site.value.context
+      if (show && activeI18n && !(activeI18n in i18n)) {
+        // Set the active language back to default if it does not exist
+        // in the editor context. This is most likely to occur when the
+        // language is removed using the undo command.
+        const defaultLanguage = currentLanguages.value.find(
+          (language) => language.value === DEFAULT_LANGUAGE,
+        )
+        if (defaultLanguage) {
+          selectActiveLanguage(defaultLanguage)
+        }
```

Comment:
Hi @sampullman 

I noticed that the `lang` variable is not being used anywhere, so I removed it in this PR. Still want to double check if this make sense to you!

I also add some extra logic here to reset the active language back to default upon opening the TranslationModal, if the selected language does not exist anymore:

https://github.com/samatechtw/pubstudio-builder/assets/59676941/aa6651ed-47ee-4fb2-884d-b4aeba3664b0